### PR TITLE
Added keyboard shortcuts

### PR DESCRIPTION
**Pull request by paulsbduncan**

### CODE REVIEW

**Comment by blairconrad** [comment]
Hey, @paulsbduncan. Thanks for your interest! I'll review the changes as soon as I'm able.

File: Extension/background.js [hunk 1]
```diff
@@ -2,6 +2,23 @@
...
+function copyToClipboard(text) {
+  var ta = document.getElementById('ta');
+  ta.style.display = 'block';
+  ta.value = text;
+  ta.select();
+  document.execCommand('copy');
+  ta.style.display = 'none';
+}
+
+function snagTitle(response){
```
Comment (blairconrad) [inline Extension/background.js]: nit: outside of the original context (where it was specified as the object sent back by a callback), the name `response` isn't very informative. Perhaps something that conveys that it's information about an issue?

File: Extension/background.js [hunk 2]
```diff
@@ -2,6 +2,23 @@
...
+  copyToClipboard(response.issueNumber + ' - ' + response.title);
+}
+
+function snagLink(response){
+  copyToClipboard('[' + response.issueNumber + ' - ' + response.title + '](http://jiraprod.agfahealthcare.com/browse/' + response.issueNumber + ')');
+}
```
Comment (blairconrad) [inline Extension/background.js]: I like the snag* methods.

File: Extension/background.js [hunk 3]
```diff
@@ -24,3 +41,25 @@ chrome.runtime.onInstalled.addListener(function() {
...
+      alert('There was an error injecting script: \n' + chrome.runtime.lastError.message);
+    }
+  });
+
+  chrome.tabs.query({active: true, currentWindow: true}, function(tabs) {
+    var tabId=tabs[0].id;
+    chrome.tabs.sendMessage(tabId, {action: 'get page details'}, function(response) {
+      if( command === "copy-title" ) {
+        snagTitle(response);
+      } else if( command == "copy-link" ) {
```
Comment (blairconrad) [inline Extension/background.js]: Why `===` above and `==` here (and below)?

**Review by blairconrad — CHANGES_REQUESTED** [review]
I 😍 this change so much!
It works beautifully.

I've left a couple of questions and a nit.

There are also some minor formatting inconsistencies, like missing whitespace near `{`s and `(`s, but let's let those slide.

**Comment by paulsbduncan** [comment]
Thanks.  I've made the changes you suggested and pushed them up again.  What do I do now?  Another pull request, or is this more work on this existing pull request?

**Comment by paulsbduncan** [comment]
I see above that it says "Add more commits by pushing to the keyboard-shortcuts branch on paulsbduncan/AgfaJiraChromeExtension."  I did that.  Is that it from me?

**Comment by blairconrad** [comment]
> Thanks. I've made the changes you suggested and pushed them up again. What do I do now? Another pull request, or is this more work on this existing pull request?

You're good. The pull request notices that the branch has been updated, and we (you, me, everyone) can see the new content. Check the **Commits** or **Files changed** tabs upstairs to verify.

**Review by blairconrad — DISMISSED** [review]

**Comment by blairconrad** [comment]
Thanks, @paulsbduncan. It all looks great. One thing I'd consider is squashing the two commits, as the second one isn't really a logical change on its own. Are you interested in that, or would you like me to do it?

**Comment by paulsbduncan** [comment]
How about I watch you do it?

**Comment by blairconrad** [comment]
Sounds like you want to learn. Which will go better if you do it, because the flow is different for me (it's not my PR, after all). I'll visit. But here's the script. It's super simple.

On your keyboard-shortcuts branch, check your changes since master:

```
λ  git log master..
commit c072fbc61cb41ddcff53fc67758a0009680fba00
Author: Paul Duncan <paulsbduncan@gmail.com>
Date:   Wed May 17 07:27:22 2017 -0400

    Minor fixes after code review from Blair

commit 8ad4b76daf3f259be89d2c4577594e0337290d60
Author: Paul Duncan <paulsbduncan@gmail.com>
Date:   Tue May 16 17:03:55 2017 -0400

    Added keyboard shortcuts
```

We don't like the newer one (which is the first in the list), and wish it were part of the other one. So we'll rebase:

```
git rebase -i master
```

this just means you want to run an interactive (the `-i`) rebase session, updating your commits relative master. Now master is already the same as it was when you branched, so without the `-i`, this would be a no-op. The interactive portion will let you slice and dice the commits.

We'll see something like this in your editor (assuming you have a good `EDITOR` environment variable:

```
pick 8ad4b76 Added keyboard shortcuts
pick c072fbc Minor fixes after code review from Blair

# Rebase 989e3b3..c072fbc onto 989e3b3 (2 command(s))
#
# Commands:
# p, pick = use commit
# r, reword = use commit, but edit the commit message
# e, edit = use commit, but stop for amending
# s, squash = use commit, but meld into previous commit
# f, fixup = like "squash", but discard this commit's log message
# x, exec = run command (the rest of the line) using shell
# d, drop = remove commit
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
# Note that empty commits are commented out
```

Now we do want to keep (or "pick") the first commit, but the second one was really just a fixup of the first, so we amend:

```
pick 8ad4b76 Added keyboard shortcuts
fixup c072fbc Minor fixes after code review from Blair

…
```

Save and exit the editor.

This will spew something like this on the console:

```
λ  git rebase -i master
[detached HEAD 0942440] Added keyboard shortcuts
 Author: Paul Duncan <paulsbduncan@gmail.com>
 Date: Tue May 16 17:03:55 2017 -0400
 5 files changed, 72 insertions(+), 13 deletions(-)
 create mode 100644 Extension/background.html
Successfully rebased and updated refs/heads/keyboard-shortcuts.
```

and we check the log again:

```
λ  git log master..
commit 09424401cff70409c620e562a33bc98b21f28f37
Author: Paul Duncan <paulsbduncan@gmail.com>
Date:   Tue May 16 17:03:55 2017 -0400

    Added keyboard shortcuts
```

Now there's just one commit (with a new commit hash!).

And we can push it up to your branch on GitHub. Because you're rewriting commits, not just appending, you'll need to force. You can do this like

```
git push --force origin keyboard-shortcuts
```

Or just do it in GitKraken:

image

and pick "Force Push":

image

**Review by blairconrad — APPROVED** [review]
Smashing!

**Comment by blairconrad** [comment]
This issue has been released in version 0.3.0.

@paulsbduncan, look for your name in the [release notes](https://github.com/blairconrad/AgfaJiraChromeExtension/releases/tag/0.3.0)! :trophy:

**Pull request merged**